components/lwip/port/esp32/include/netinet/tcp.h

```diff
@@ -0,0 +1,21 @@
+// Copyright 2015-2016 Espressif Systems (Shanghai) PTE LTD
+// Copyright 2020 Francesco Giancane <francesco.giancane@accenture.com>
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef TCP_H_
+#define TCP_H_
```

Comment:
This is minor, but `TCP_H_` seems like a generic enough name that there's a chance of a conflict with another header at some point.

What about `_NETINET_TCP_H` (this is what the BSD headers use) or switching over to `#pragma once` instead?

Reply:
Hi @projectgus ,

You are right. Indeed I mimicked the other _compatibility_ headers but I find your `_NETINET_TCP_H` more appropriate.

For what concerns philosophical reasons, please have a look at https://github.com/espressif/esp-idf/tree/master/components/lwip/port/esp32/include 
It seems that lwIP already assumes that compatibility headers are provided as a final integration step.

(in fact, `esp-idf` already provided `netinet/in.h`)

Reply:
Thanks. I've squashed the rename commit and pushed this into our internal review and merge queue, the PR will be updated when it merges.

Regarding philosophy, this is something we've done for compatibility reasons in the past but the upstream also contains a "compat" header tree which is more sparse:
https://git.savannah.nongnu.org/cgit/lwip.git/tree/src/include/compat/posix

I don't know if there's a hard-and-fast rule about what should be here and what is the responsibility of the port. But I'm happy to merge this into our port in any case.

Reply:
@projectgus ,

Thank you for your feedback.
For the philosophy point, I am also on the side that compatibility should be done by lwIP and not by port side, however if you look at their compat layer, `tcp.h` is still missing.

Probably we should ask lwIP developers directly, but I think that in the while we can live happily with this PR :)